### PR TITLE
Add decorator common.utils.exception_handler for rc_load_… functions

### PR DESCRIPTION
**Pull request by mediapills**
## High-level description

Add decorator common.utils.exception_handler for rc_load_json, rc_load_ini and rc_load_yaml for better maintainability and decrease code duplication.

## Corresponding DC/OS tickets (required)

  - [D2IQ-65069](https://jira.d2iq.com/browse/D2IQ-65069) Add decorator common.utils.exception_handler for rc_load_json, rc_load_ini and rc_load_yaml.

## Related tickets (optional)

  - [D2IQ-ID](https://jira.mesosphere.com/browse/D2IQ-<number>) JIRA title / short description.

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [ ] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [ ] Included a test which will fail if code is reverted but test is not. If there is no test please explain.
  - [ ] Test Results: [link to CI job test results for component]
  - [ ] Code Coverage (if available): [link to code coverage report]

### CODE REVIEW

**Comment by mediapills** [comment]
@mesosphere-mergebot request-review @jongiddy

**Comment by mediapills** [comment]
@mesosphere-mergebot changelog-not-required Fix to unreleased code

File: packages/winpanda/extra/src/winpanda/core/utils.py [hunk 1]
```diff
@@ -135,45 +120,20 @@ def rc_load_yaml(fpath, emheading=None, render=False, context=None):
...
-                y_body = yaml.safe_load(fp)
-    except (FileNotFoundError, jj2.TemplateNotFound) as e:
-        err_msg = f'Load: {fpath}'
-        err_msg = f'{emheading}: {err_msg}' if emheading else err_msg
-        raise cr_exc.RCNotFoundError(err_msg) from e
-    except (OSError, RuntimeError) as e:
-        err_msg = f'Load: {fpath}: {type(e).__name__}: {e}'
-        err_msg = f'{emheading}: {err_msg}' if emheading else err_msg
-        raise cr_exc.RCError(err_msg) from e
-    except (jj2.TemplateError, yaml.YAMLError) as e:
```
Comment (jongiddy) [inline packages/winpanda/extra/src/winpanda/core/utils.py]: `yaml.YAMLError` is not caught by the decorator.

File: packages/winpanda/extra/src/winpanda/core/utils.py [hunk 2]
```diff
@@ -17,6 +17,43 @@
...
+            err_msg = f'{emheading}: {err_msg}' if emheading else err_msg
+            raise cr_exc.RCError(err_msg) from e
+        except (jj2.TemplateError,) + cr_exc.JSON_ERRORS as e:
+            err_msg = f'Load: {fpath}: {type(e).__name__}: {e}'
+            err_msg = f'{emheading}: {err_msg}' if emheading else err_msg
+            raise cr_exc.RCInvalidError(err_msg) from e
+        except (jj2.TemplateError, yaml.YAMLError) as e:
+            err_msg = f'Load: {fpath}: {type(e).__name__}: {e}'
+            err_msg = f'{emheading}: {err_msg}' if emheading else err_msg
+            raise cr_exc.RCInvalidError(err_msg) from e
```
Comment (jongiddy) [inline packages/winpanda/extra/src/winpanda/core/utils.py]: These can be combined into a single `(jj2.TemplateError, yaml.YAMLError) + cr_exc.JSON_ERRORS` clause. The bodies are the same and the second mention of `jj2.TemplateError` has no effect.

File: packages/winpanda/extra/src/winpanda/core/utils.py (hunk 1 above)
Comment (mediapills) [reply packages/winpanda/extra/src/winpanda/core/utils.py]: Tests added, exception handler added in decorator

**Comment by DominikDary** [comment]
@mesosphere-mergebot changelog-not-required internal change

**Pull request merged**